clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/TriggersMatcher.kt

```diff
@@ -0,0 +1,404 @@
+     *
+     * Conditions are OR-ed; returns true if any condition is satisfied.
+     *
+     * @param trigger The [TriggerAdapter] having trigger condition to be matched against the event.
+     * @param event The [EventAdapter] having event to be matched against the trigger condition.
+     * @return True if user location matches any geo-radius condition; otherwise, false.
+     */
+    @VisibleForTesting
+    internal fun matchGeoRadius(event: EventAdapter, trigger: TriggerAdapter): Boolean {
+        if (event.userLocation != null && event.userLocation.isValid()) {
+            // GeoRadius conditions are OR-ed
+            for (i in 0 until trigger.geoRadiusCount) {
+                val triggerRadius = trigger.geoRadiusAtIndex(i)
+                val expected = Location("")
+                expected.latitude = triggerRadius!!.latitude
+                expected.longitude = triggerRadius.longitude
+
+                try {
+                    if (evaluateDistance(triggerRadius.radius, expected, event.userLocation)) {
+                        return true
+                    }
+                } catch (e: Exception) {
+                    Logger.d("Error matching GeoRadius triggers for event named ${event.eventName}. Reason: ${e.localizedMessage}")
+                }
+            }
+        }
+        return false
+    }
+
+    /**
+     * Internal function to evaluate a trigger condition.
+     *
+     * This function evaluates a single trigger condition by comparing the expected value with the actual value.
+     * The specified operator (`op`) is used to perform the comparison.
+     *
+     * @param op The [TriggerOperator] used for comparison.
+     * @param expected The expected [TriggerValue] for the condition.
+     * @param actual The actual [TriggerValue] to be compared with the expected value.
+     * @return `true` if the condition is satisfied, `false` otherwise.
+     */
+    @VisibleForTesting
+    internal fun evaluate(
+        op: TriggerOperator,
+        expected: TriggerValue,
+        actual: TriggerValue
+    ): Boolean {
+        if (actual.value == null) {
+            return op == TriggerOperator.NotSet
+        }
+
+        return when (op) {
+            TriggerOperator.Set -> true
+            TriggerOperator.LessThan -> expectedValueLessThanGreaterThanActual(expected, actual, true)
+            TriggerOperator.GreaterThan -> expectedValueLessThanGreaterThanActual(expected, actual, false)
+            TriggerOperator.Equals -> expectedValueEqualsActual(expected, actual)
+            TriggerOperator.NotEquals -> !expectedValueEqualsActual(expected, actual)
+            TriggerOperator.Between -> actualIsInRangeOfExpected(expected, actual)
+            TriggerOperator.Contains -> actualContainsExpected(expected, actual)
+            TriggerOperator.NotContains -> !actualContainsExpected(expected, actual)
+            else -> false // TODO: Implement all cases as per the backend evaluation and remove this line
```

Comment:
lets have exhaustive when and resolve this todo

Reply:
@piyush-kukadiya